src/main/kotlin/com/zepben/evolve/cim/iec61970/base/core/NameType.kt

```diff
@@ -104,9 +139,34 @@ class NameType(val name: String) {
      *
      * @return true if a matching name was removed.
      */
-    fun removeNames(name: String?): Boolean = (namesIndex.remove(name) ?: namesMultiIndex.remove(name)) != null
+    fun removeNames(name: String?): Boolean {
+        // Calling removeName from identifiedObject will remove the name from both the identifiedObject and the nameType
+        namesIndex[name]?.let { n ->
+            return n.identifiedObject.removeName(n)
+        }
+        namesMultiIndex[name]?.values.let { names ->
+            var removed = false
+            names?.toList()?.forEach { name ->
+                removed = name.identifiedObject.removeName(name) || removed
+            }
+            return removed
+        }
+    }
 
+    /**
+     * Removes all [Name] instances associated with this [NameType].
+     *
+     * @return this [NameType]
+     */
     fun clearNames(): NameType {
+        // Remove name from associated identified object when it's removed from the nameType
+        namesIndex.toMap().forEach { (_, name) ->
+            name.identifiedObject.removeName(name)
+        }
+        namesMultiIndex.flatMap { it.value.values }.forEach { name ->
+            name.identifiedObject.removeName(name)
```

Comment:
todo: Codecov indicates this line is not covered. Can you check if `Codecov` is full of it, or add a test for `clearNames` with a multi instances name and confirm that all names are correctly removed from their respective objects.

Reply:
Done